### PR TITLE
TsLint migrate to EsLint

### PR DESCRIPTION
**Pull request by DimalChandrasiri**
## Purpose
This PR will migrate to `eslint` for lint purposes since the `tslint` is deprecated and migrated to `typescript-eslint`.

### CODE REVIEW

File: apps/admin-portal/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    extends: [
+        "../../.eslintrc.js"
+    ]
+};
```
Comment (jeradrutnam) [inline apps/admin-portal/.eslintrc.js]:
```suggestion
};

```

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    settings: {
+        react: {
+            version: "detect"
+        },
+    },
+    rules: {
+        // Place to specify ESLint rules. Can be used to overwrite rules specified from the extended configs
+        // e.g. "@typescript-eslint/explicit-function-return-type": "off",
+    },
+};
```
Comment (jeradrutnam) [inline .eslintrc.js]:
```suggestion
};

```

File: apps/admin-portal/src/api/profile.ts [hunk 3]
```diff
@@ -162,23 +182,7 @@ export const updateUserInfo = (userId: string, data: object): Promise<any> => {
...
-            .get(url)
-            .then((response) => {
-                resolve(url.split("?")[0]);
-            })
-            .catch((error) => {
-                reject();
-            });
-    });
-};
+
```
Comment (jeradrutnam) [inline apps/admin-portal/src/api/profile.ts]:
```suggestion
```

File: apps/admin-portal/src/locales/en/views.ts [hunk 4]
```diff
@@ -390,22 +390,22 @@ export const views: Views = {
...
+                    nameGivenName: "First name",
                     phoneNumbers: "Phone number",
-                    phoneNumbers_home: "Home phone number",
-                    phoneNumbers_mobile: "Mobile number",
-                    phoneNumbers_other: "Other phone number",
-                    phoneNumbers_work: "Work phone number",
+                    phoneNumbersHome: "Home phone number",
+                    phoneNumbersMobile: "Mobile number",
+                    phoneNumbersOther: "Other phone number",
+                    phoneNumbersWork: "Work phone number",
```
Comment (jeradrutnam) [inline apps/admin-portal/src/locales/en/views.ts]: Please double check this. As these values are coming from the scim meta endpoint. So these are the dynamic key names that receives. Please check the localization in User Portal profile details page ad confirm.

File: apps/user-portal/.eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    extends: [
+        "../../.eslintrc.js"
+    ]
+};
```
Comment (jeradrutnam) [inline apps/user-portal/.eslintrc.js]:
```suggestion
};
```
```suggestion
};

```

Comment (jeradrutnam) [inline apps/user-portal/.eslintrc.js]:
```suggestion
};

```

File: apps/user-portal/src/api/multi-factor-fido.ts [hunk 6]
```diff
@@ -279,6 +169,40 @@ const responseToObject = (response) => {
...
+                );
+            }
+            return Promise.resolve(response);
+        })
+        .catch((error) => {
+            return Promise.reject(`Failed to finish the FIDO registration - ${error}`);
+        });
+};
+
+
```
Comment (jeradrutnam) [inline apps/user-portal/src/api/multi-factor-fido.ts]:
```suggestion
```

File: apps/admin-portal/src/locales/en/views.ts (hunk 4 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/locales/en/views.ts]: Smoke tested the Profile details page,  and looks fine.

File: modules/authentication/.eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    extends: [
+        "../../.eslintrc.js"
+    ]
+};
```
Comment (jeradrutnam) [inline modules/authentication/.eslintrc.js]:
```suggestion
};

```

File: apps/user-portal/src/index.tsx [hunk 8]
```diff
@@ -16,7 +16,7 @@
  * under the License.
  */
 
-import { ContextUtils } from "@wso2is/core/utils";
+import { ContextUtils, } from "@wso2is/core/utils";
```
Comment (jeradrutnam) [inline apps/user-portal/src/index.tsx]:
```suggestion
import { ContextUtils } from "@wso2is/core/utils";
```

File: modules/authentication/src/models/sign-in.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jeradrutnam) [inline modules/authentication/src/models/sign-in.ts]:
```suggestion
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: modules/authentication/src/models/sign-in.ts [hunk 10]
```diff
@@ -16,23 +16,22 @@
...
-    refresh_token: "",
-    username: ""
-});
+export interface TokenRequestHeader {
+    headers: {
+        Accept: string,
+        "Access-Control-Allow-Origin": string,
+        "Content-Type": string
+    }
+}
```
Comment (jeradrutnam) [inline modules/authentication/src/models/sign-in.ts]:
```suggestion
}

```

Comment (jeradrutnam) [inline modules/authentication/src/models/sign-in.ts]: Why this file is changed?

File: modules/core/.eslintrc.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    extends: [
+        "../../.eslintrc.js"
+    ]
+};
```
Comment (jeradrutnam) [inline modules/core/.eslintrc.js]:
```suggestion
};

```

File: modules/forms/.eslintrc.js [hunk 12]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    extends: [
+        "../../.eslintrc.js"
+    ]
+};
```
Comment (jeradrutnam) [inline modules/forms/.eslintrc.js]:
```suggestion
};

```

File: modules/http/.eslintrc.js [hunk 13]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    extends: [
+        "../../.eslintrc.js"
+    ]
+};
```
Comment (jeradrutnam) [inline modules/http/.eslintrc.js]:
```suggestion
};

```

File: modules/react-components/.eslintrc.js [hunk 14]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    extends: [
+        "../../.eslintrc.js"
+    ]
+};
```
Comment (jeradrutnam) [inline modules/react-components/.eslintrc.js]:
```suggestion
};

```

File: modules/theme/.eslintrc.js [hunk 15]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    extends: [
+        "../../.eslintrc.js"
+    ]
+};
```
Comment (jeradrutnam) [inline modules/theme/.eslintrc.js]:
```suggestion
};

```

File: modules/validation/.eslintrc.js [hunk 16]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    extends: [
+        "../../.eslintrc.js"
+    ]
+};
```
Comment (jeradrutnam) [inline modules/validation/.eslintrc.js]:
```suggestion
};

```

**Pull request merged**